hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java

```diff
@@ -179,6 +201,87 @@ private void extractAndReplaceArtifacts(
         }
     }
 
+    private void generateConfigPem(@NonNull final Path artifactsLoc) {
+        requireNonNull(artifactsLoc, "Cannot generate config.txt without a valid artifacts location");
+
+        final var configTxt = artifactsLoc.resolve("config.txt");
+        final var nodeState = nodeStore.nodesState();
+
+        if (nodeState == null || nodeState.size() == 0) {
+            log.info("Node state is empty, cannot generate config.txt"); // change to log error later
+            return;
+        }
+
+        final var nodeIds = nodeStore.nodesState().keys();
```

Comment:
We should not give back nodeState from the store. Just return the keys.

Reply:
I also need to setup unit test here by:
given(nodeStore.nodesState()).willReturn(readableNodeState);
in class ReadableFreezeUpgradeActionsTest.

Reply:
Changed to use keys().